src/main/java/hudson/remoting/AbstractByteBufferCommandTransport.java

```diff
@@ -199,10 +199,8 @@ private void processCommand() throws IOException {
         try {
             FastByteBufferQueueInputStream is = new FastByteBufferQueueInputStream(receiveQueue, readCommandSizes[0]);
             try {
-                ObjectInputStreamEx ois = new ObjectInputStreamEx(is, channel.baseClassLoader, channel.classFilter);
-                Command cmd = Command.readFrom(channel, ois);
+                Command cmd = Command.readFrom(channel, is, readCommandSizes[0]);
                 receiver.handle(cmd);
-                channel.notifyRead(cmd, readCommandSizes[0]);
```

Comment:
@jglick I believe here was a glitch in the handling. Read notification should have happened before handling. Fixed by the patch. Please correct me if I miss something.

Reply:
Not sure it really matters, but did you test it? (Integration test in `support-core`.) Anyway we only cared about `NioTransport`s.

Reply:
Testing